File: java/src/main/java/com/example/infraestructure/adapters/controllers/TestController.java
package com.example.infraestructure.adapters.controllers;

import com.example.application.dto.ResponseDto;
import com.example.application.use_cases.TestPostUseCase;
import com.example.application.use_cases.TestUseCase;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.io.IOException;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;

@RestController
@RequestMapping("/api/v1/test")
@AllArgsConstructor
public class TestController {

    private final TestUseCase useCase;
    private final TestPostUseCase testPostUseCase;

    @GetMapping("post")
    public ResponseEntity<ResponseDto> post() throws NoSuchAlgorithmException, IOException, InvalidKeyException, InterruptedException {
        return new ResponseEntity<>(testPostUseCase.execute(), HttpStatus.OK);
    }

    @GetMapping("get")
    public ResponseEntity<ResponseDto> get() throws NoSuchAlgorithmException, IOException, InvalidKeyException, InterruptedException {
        return new ResponseEntity<>(useCase.execute(), HttpStatus.OK);
    }
}

File: nodejs/src/infrastructure/adapters/controllers/test.controller.ts
import { Controller, Get } from '@nestjs/common';
import {TestUseCase} from "../../../application/use-cases/test.use-case";
import {TestPostUseCase} from "../../../application/use-cases/test-post.use-case";

@Controller('test')
export class TestController {
  constructor(private readonly testUseCase: TestUseCase,
              private readonly postUseCase: TestPostUseCase) {}

  @Get('/get')
  handle() {
    return this.testUseCase.execute();
  }

  @Get('/post')
  handlePost() {
    return this.postUseCase.execute();
  }
}
